app/javascript/chat/__tests__/reportAbuse.test.jsx

```diff
@@ -0,0 +1,44 @@
+import { h } from 'preact';
+import { render } from '@testing-library/preact';
+import { axe } from 'jest-axe';
+import ReportAbuse from '../ReportAbuse';
+
+describe('<ReportAbuse />', () => {
```

Comment:
I'm really curious about this as well, previously I have handled those kind of actions in jest, but that was honestly due to a lack of an E2E testing framework. But...my gut says this is a great candidate for an E2E test, yeah!

Reply:
Yeah, once I've wrapped up the Cypress data cleaning, this definitely merits an E2E test. It's a more long term thing, but it'd be good to map out key flows in the app that we want to add end to end tests for. cc: @vaidehijoshi .

Reply:
> It's a more long term thing, but it'd be good to map out key flows in the app that we want to add end to end tests for. 

@nickytonline Yeah, this is certainly on my mind; specifically, I've been chatting with @lisasy about creating a full representation of _all_ the flows in our app, just from a _product_ perspective. 

@citizen428 actually gave me the idea of laying out every single screen and state that exists in a Forem, and using that to get a sense of how far-reaching the tentacles of our product really are. The benefit of this being, of course, that those flows literally map to the E2E flows that we'd want to test across the platform.

Suffice it to say, it will be somewhat of an undertaking, and step 1 is just figuring out what the flows are. But it would take us towards this ultimate north start of E2E tests across the key flows of the app, for sure :)